recommender/recommender.py

```diff
@@ -368,15 +370,18 @@ def _check_upload_file(self, request, file_types, file_type_error_msg, event, fi
                 if 'magic' in file_types[file_type]:
                     # Check magic number
                     headers = file_types[file_type]['magic']
-                    if request.POST['file'].file.read(len(headers[0]) / 2).encode('hex') not in headers:
+                    file_pointer = request.POST['file'].file
+                    file_magic_number = file_pointer.read(int(len(headers[0]) / 2))
+
+                    if codecs.encode(file_magic_number, 'hex').decode('utf-8') not in headers:
                         file_type_error = True
                     request.POST['file'].file.seek(0)
 
         if file_type_error:
             response = Response()
             tracker.emit(event, {'uploadedFileName': 'FILE_TYPE_ERROR'})
             response.status = 415
-            response.body = json.dumps({'error': file_type_error_msg})
+            response.text = json.dumps({'error': file_type_error_msg})
```

Comment:
If you want to provide string objects, you provide is as text. And `body` is for byte objects in python 3.